internal/services/dispatcher/server.go

```diff
@@ -549,8 +549,7 @@ func (s *DispatcherImpl) Heartbeat(ctx context.Context, req *contracts.Heartbeat
 		span.RecordError(err)
 		span.SetStatus(telemetry_codes.Error, "could not get worker")
 		if errors.Is(err, pgx.ErrNoRows) {
-			s.l.Error().Msgf("worker %s not found", req.WorkerId)
-			return nil, err
+			return nil, fmt.Errorf("worker %s not found", req.WorkerId)
```

Comment:
This is still returning a 500-level error to the user. We want to return a 404 or something similar

Reply:
ah whoops, missed that 🤦 thanks